lib/elixir/lib/string.ex

```diff
@@ -240,37 +240,48 @@ defmodule String do
       iex> String.printable?("abc" <> <<0>>, 2)
       true
 
+      iex> String.printable?("abc" <> <<0>>, 0)
+      true
+
   """
-  @spec printable?(t) :: boolean
-  @spec printable?(t, non_neg_integer | :infinity) :: boolean
-  def printable?(string, counter \\ :infinity)
+  @spec printable?(t, 0) :: true
+  @spec printable?(t, pos_integer | :infinity) :: boolean
+
+  def printable?(string, character_limit \\ :infinity)
+      when is_binary(string) and
+             (character_limit == :infinity or
+                (is_integer(character_limit) and character_limit >= 0)) do
+    do_printable?(string, character_limit)
```

Comment:
I believe we're trying to not have `do_` prefixing, I wonder if we can come up with an alternative. 🤔

Reply:
I could not come up with anything, but this is a common pattern. What can we name functions that take the same arguments as the public one, but we don't do any guard checks.
`printable?_checked`, `printable?_filtered`, `printable?_guarded` ???

Reply:
I thought the same but I have no good suggestion either.

Reply:
Maybe `printable_with_limit`?

Reply:
Sorry, but that doesn't say anythig to me as the public one uses limit as well.
How about process_printable?, Or pass_, consume_, I'm sure somebody else can find a better verb to describe this action

Reply:
All the prefixed alternatives share the same issue as the `do_`—breaks grouping and navigation. Ideally we should go with suffix. What do you folks think about `printable_impl`? Still not ideal but at least helps to solve aforementioned issue.